configs/templates/generated_interfaces_for_adapters/list/ListProcessor.go

```diff
@@ -0,0 +1,14 @@
+package list
+
+import "istio.io/mixer/configs/templates/list"
+
+type Instance struct {
+  Blacklist       bool
+  CheckExpression interface{}
+}
+
+type ListProcessor interface {
+  ConfigureList(templateName string, types map[string]*foo_bar_mylistchecker.Type /*typeName to Type mapping*/) error
+  CheckList(templateName string, instances map[string]*Instance /*typeName to Instance (generated from Constructor) mapping*/) (bool, error)
```

Comment:
Converting the struct back to your type shouldn't be expensive. IIRC the code generated in Struct amounts to a normal golang type assertion (`myType, ok := struct.(myType)`). I really really want to avoid us exposing proto.Value to adapter authors, it is not ergonomic in go. In my mind the whole point of us generating code is that we can make it type-safe and provide a better interface for adapter authors. If we're going to do generic stuff then why bother doing code gen? We could pass in a proto.Struct today.